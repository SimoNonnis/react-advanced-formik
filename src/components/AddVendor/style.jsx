import styled from 'styled-components';
import { media } from '@housesimple/react-components';

export const AddVendorContainer = styled.div`
  text-align: right;
  padding: ${({ theme }) => theme.spacing.md} 0;

  ${media.tablet`
    padding: ${({ theme }) => theme.spacing.lg} 0;
  `};
`;
