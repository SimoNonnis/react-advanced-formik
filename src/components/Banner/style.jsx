import styled from 'styled-components';
import { Attention } from '@housesimple/react-components';

export const Container = styled.div`
  ${({ theme: { color }, type }) => type === 'user' && `background-color: ${color.red15}`};
  ${({ theme: { color }, type }) => type === 'system' && `background-color: ${color.lightOrange}`};
  ${({ theme: { color }, type }) => type === 'success' && `background-color: ${color.lightGreen}`};
  padding: ${({ theme }) => theme.spacing.md};
  position: fixed;
  top: 10%;
  right: 5%;
  z-index: 3;
  max-width: 80%;
  text-align: center;
`;

export const AttentionIcon = styled(Attention)`
  ${({ theme: { color }, type }) => type === 'user' && `fill: ${color.red}`};
  ${({ theme: { color }, type }) => type === 'system' && `fill: ${color.veryDarkBlue}`};
  ${({ theme: { color }, type }) => type === 'success' && `fill: ${color.veryDarkBlue}`};
  margin-right: 6px;
  vertical-align: sub;
`;
