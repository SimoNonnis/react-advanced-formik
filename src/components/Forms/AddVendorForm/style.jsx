import styled from 'styled-components';
import { H2 } from '@housesimple/react-components';

export const TitleContainer = styled.div`
  margin-bottom: ${({ theme }) => theme.spacing.lg};
`;

export const Title = styled(H2)`
  font-size: 27px;
  margin: 0;
`;
