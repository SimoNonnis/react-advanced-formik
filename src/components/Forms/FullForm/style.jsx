import styled from 'styled-components';
import { media, Avatar, H2 } from '@housesimple/react-components';

export function truncate(width) {
  return `
    width: ${width};
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  `;
}

export const Head = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const VendorInfo = styled.div`
  display: flex;
  align-items: center;
`;

export const VendorIcon = styled(Avatar)`
  color: ${({ theme }) => theme.color.mediumGrey};
`;

export const VendorFullName = styled(H2)`
  ${truncate('115px')};
  ${media.phone`
    ${truncate('250px')};
  `};
  ${media.smallTablet`
    ${truncate('350px')};
  `};

  margin: 0;
  margin-left: ${({ theme }) => theme.spacing.sm};
  padding: 7px 0;
`;

export const VendorLastName = styled.span`
  display: inline-block;
  padding-left: ${({ theme }) => theme.spacing.xs};
`;

export const Body = styled.div`
  margin-top: ${({ theme }) => theme.spacing.md};

  ${media.tablet`
    margin-top: ${({ theme }) => theme.spacing.lg};
  `};
`;
