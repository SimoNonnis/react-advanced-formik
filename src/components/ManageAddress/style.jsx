import styled from 'styled-components';
import { media, Button } from '@housesimple/react-components';

export const Container = styled.div`
  position: relative;
`;

export const AddAddressContainer = styled.div`
  margin: ${({ theme }) => theme.spacing.md} 0;
  text-align: center;
`;

export const AddressContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  ${media.phone`
    flex-direction: row;
  `};

  padding: ${({ theme }) => theme.spacing.md};
  border: 1px solid ${({ theme }) => theme.color.lightGrey2};
  & + & {
    margin-top: ${({ theme }) => theme.spacing.md};
  }
`;

export const DeleteButton = styled(Button)`
  color: ${({ theme }) => theme.color.red};

  ${media.phone`
    font-size: 14px;
  `};

  padding: ${({ theme }) => theme.spacing.sm};
  text-decoration: none;

  &:hover {
    color: ${({ theme }) => theme.color.red};
    text-decoration: underline;
  }
`;
