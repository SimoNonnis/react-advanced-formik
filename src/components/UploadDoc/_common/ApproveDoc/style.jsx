import styled from 'styled-components';
import {
  media,
  DocumentPlaceholder,
  P,
  RadioBox,
  Button as Btn,
} from '@housesimple/react-components';

export const PreviewContainer = styled.div`
  margin-top: ${({ theme }) => theme.spacing.lg};
  background-color: ${({ theme }) => theme.color.lightGrey3};
  padding: ${({ theme }) => theme.spacing.md};
  padding-bottom: ${({ theme }) => theme.spacing.sm};
  display: flex;
  flex-direction: column;
  border-radius: 4px;
  justify-content: center;
  align-items: center;
`;

export const Preview = styled.div`
  border-radius: 4px;
  background-color: ${({ theme }) => theme.color.white};
  padding: ${({ theme }) => theme.spacing.xs};
  color: ${({ theme }) => theme.color.mediumBlue};
`;

export const FileName = styled(P)`
  margin: 0;
  margin-top: ${({ theme }) => theme.spacing.xs};
`;

export const DocPlaceholder = styled(DocumentPlaceholder)`
  display: block;
`;

export const FilePreview = styled.img`
  display: block;
  width: 210px;
  height: 150px;
  object-fit: cover;

  ${media.phone`
    width: 250px;
    height: 190px;
  `};
`;

export const TypeOfDocContainer = styled.div`
  margin: ${({ theme }) => theme.spacing.lg} 0;

  ${media.phone`
    max-width: 80%;
  `};
`;

export const ConfirmContainer = styled.div`
  margin: ${({ theme }) => theme.spacing.lg} 0;

  ${media.phone`
    max-width: 80%;
  `};
`;

export const ConfirmRadioBox = styled(RadioBox)`
  display: block;

  & + & {
    margin-top: ${({ theme }) => theme.spacing.md};
    margin-left: 0;
  }
`;

export const ConfirmLabel = styled.span`
  font-weight: 400;
`;

export const ButtonsContainer = styled.div`
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
`;

export const Button = styled(Btn)`
  min-width: 181px;
  margin: ${({ theme }) => theme.spacing.sm};
`;
