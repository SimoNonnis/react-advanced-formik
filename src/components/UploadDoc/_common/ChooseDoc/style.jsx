import styled from 'styled-components';

import { P } from '@housesimple/react-components';

export const Title = styled.h3`
  font-size: 22px;
  font-weight: 600;
  margin: 0;
  margin-bottom: ${({ theme }) => theme.spacing.sm};
`;

export const Description = styled(P)`
  margin-bottom: ${({ theme }) => theme.spacing.xl};
`;
