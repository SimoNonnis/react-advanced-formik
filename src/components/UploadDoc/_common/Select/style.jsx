import styled from 'styled-components';

export const Label = styled.label`
  color: ${({ theme }) => theme.color.veryDarkBlue};
  font-family: ${({ theme }) => theme.base.fontFamily};
  margin-bottom: 7px;
  display: block;
  font-size: ${({ theme }) => theme.base.fontSize};
  font-weight: 600;
`;

export const SelectWrap = styled.div`
  border-radius: 2px;
  border: 1px solid ${({ theme }) => theme.color.stoneGrey};
  color: ${({ theme }) => theme.color.veryDarkBlue};
  position: relative;

  &:after {
    border-left: 6px solid transparent;
    border-right: 6px solid transparent;
    border-top: 8px solid ${({ theme }) => theme.color.veryDarkBlue};
    content: '';
    height: 0;
    pointer-events: none;
    position: absolute;
    right: 20px;
    top: 50%;
    transform: translateY(-50%);
    width: 0;
    z-index: 2;
  }
`;

export const SelectStyled = styled.select`
  box-sizing: border-box;
  display: block;
  appearance: none;
  background: none;
  box-shadow: none;
  border: none;
  box-shadow: none;
  outline: 0;
  font-family: ${({ theme }) => theme.base.fontFamily};
  font-size: 16px;
  font-weight: inherit;
  height: auto;
  line-height: 24px;
  margin: 0;
  width: 100%;
  color: ${({ theme }) => theme.color.veryDarkBlue};
  padding: 13px 50px 13px 15px;
`;
