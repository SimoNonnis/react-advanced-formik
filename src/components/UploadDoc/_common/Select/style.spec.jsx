/* eslint-env jest */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { theme } from '@housesimple/react-components';

import { Label, SelectWrap, SelectStyled } from './style';

describe('Test <Label />', () => {
  it('should render Label with theme props', () => {
    const wrapper = shallow(<Label theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <SelectWrap />', () => {
  it('should render SelectWrap with theme props', () => {
    const wrapper = shallow(<SelectWrap theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('Test <SelectStyled />', () => {
  it('should render SelectStyled with theme props', () => {
    const wrapper = shallow(<SelectStyled theme={theme} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
