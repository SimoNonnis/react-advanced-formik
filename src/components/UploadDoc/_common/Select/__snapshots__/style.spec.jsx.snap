// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Label /> should render Label with theme props 1`] = `
<label
  className="sc-htpNat jBHPye"
/>
`;

exports[`Test <SelectStyled /> should render SelectStyled with theme props 1`] = `
<select
  className="sc-ifAKCX loyIEh"
/>
`;

exports[`Test <SelectWrap /> should render SelectWrap with theme props 1`] = `
<div
  className="sc-bxivhb gNWrow"
/>
`;
